src/abstractinterpretation.jl

```diff
@@ -471,14 +465,70 @@ function maybe_report_infinite_iterations!(interp::JETInterpreter, frame::Infere
     end
 end
 
+"""
+Returns true if the iteration may terminate
+"""
+function iter_may_terminate(cond::Int64, dest::Int64, code::Vector{Any}, ssavaluetypes::Vector{Any})
+    cond_code = code[cond]
+    condt = ssavaluetypes[cond]
+    t = widenconditional(condt)
```

Comment:
```suggestion
    t = widenconditional(ignorelimited(condt))
```

We need to handle `LimitedAccuracy` here, because here we're working on pre-optimization state and thus they're not widened yet.